### ISSUE TITLE
bug: Scans stuck in Pending, Null states

### ISSUE THREAD

**Issue by specters312** [yogeshojha/rengine#1228]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### TL;DR Issue
Scans are stuck in pending states and null states assumingly from celery crashes not sure how to trouble shoot or fix and will need to spend more time on it.

### Issue
So I have scans that get stuck in null and pending states. I am not sure why and need to do more troubleshooting on the celery runners as to why this is happening. What I am assuming is happening is that there is some null exception or some exception that triggers and kills the celery runners but never reinitiates it and what not so I made 2 tools to go through and kill these scans and what not. But now the problem is scans never update from pending to running after I stop null scans.

### Python script to retrieve all project names
[project_retrieve.py.txt](https://github.com/yogeshojha/rengine/files/14950326/project_retrieve.py.txt)
### Expected Behavior

1. Scans that reach a null state should die, and initiate a scan on a new domain to begin.

2. A task should be created to frequently check if scans positions have opened up to continue scanning. (probably a feature already)

### Steps To Reproduce

I am not sure how to reproduce this behavior as I just initiate scans on large targets and notice specific domains will go into a null state.

### Environment

```markdown
- reNgine: release/2.1.0
- OS: 
PRETTY_NAME="Ubuntu 22.04.3 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.3 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian ```
- Python: Python 3.10.12
- Docker Engine: `Docker version 24.0.7, build afdd53b`
- Docker Compose: `docker-compose version 1.29.2, build 5becea4c`
- Browser: Chrome `Version 123.0.6312.106 (Official Build) (64-bit)`
```

**Comment by specters312** [comment]
Figured it out
